lambdas/service/app.py

```diff
@@ -191,41 +190,18 @@ def openapi():
 })
 @app.route('/health/{keys}', methods=['GET'], cors=True)
 def health(keys: Optional[str] = None):
-    if keys == 'basic':
-        return {
-            'up': True,
-        }
-    elif keys == 'cached':
-        storage_service = StorageService()
-        health_object = json.loads(storage_service.get('health/service'))
-        max_age = 2 * 60
-        if time.time() - health_object['time'] > max_age:
-            return Response(status_code=500,
-                            body=json.dumps({'Message': 'Cached health object is stale'}))
-        else:
-            body = health_object['health']
-    else:
-        health = Health('service')
-        if keys is None:
-            if app.current_request.context['path'].endswith('/'):
-                keys = ()
-        else:
-            keys = keys.split(',')
-        try:
-            body = health.as_json(keys)
-        except RequirementError:
-            raise BadRequestError('Invalid health keys')
-    return Response(body=json.dumps(body),
-                    status_code=200 if body['up'] else 503)
-
-
-@app.schedule('rate(1 minute)', name=config.service_cache_health_lambda_basename)
-def generate_health_object(event_: chalice.app.CloudWatchEvent):
-    storage_service = StorageService()
-    health = Health('service')
-    health = health.as_json()
-    health_object = dict(time=time.time(), health=health)
-    storage_service.put(object_key='health/service', data=json.dumps(health_object))
+    context_route_path = app.current_request.context['path']
+    health_handler = HealthHandler('service', keys, context_route_path)
+    return health_handler.response()
+
+
+@app.schedule('rate(1 minute)', name=config.cache_health_lambda_basename)
+def generate_health_object(_event: chalice.app.CloudWatchEvent):
+    service_health_handler = HealthHandler('service')
+    service_health_handler.generate_cache()
+
+    indexer_health_handler = HealthHandler('indexer')
```

Comment:
I'd like there to be a second, separate scheduled lambda for the indexer. We've chosen to implement health symmetrically and until we change our mind we should stick with that decision. Other than being slightly less efficient, do you see other problems with the symmetric approach?

Reply:
Not at all.